File: examples/documents/query-logical.swift
import JsBaoClient

// Combine conditions with logical operators ($or, $and).
func queryLogical() throws {
  // #region example
  let result = try Task.query([
    "$or": [
      ["priority": 3],
      ["dueDate": ["$lt": "2026-06-02T00:00:00Z"]],
    ],
  ])
  // #endregion example
  _ = result
}

File: dev-docs/snippets/model-surface/query-logical.ts
import { Task } from "../../../examples/_harness/generated/ts/Task.generated";

// Combine conditions with logical operators ($or, $and).
export async function queryLogical() {
  // #region example
  const result = await Task.query({
    $or: [
      { priority: 3 },
      { dueDate: { $lt: new Date().toISOString() } },
    ],
  });
  // #endregion example
  return result.data;
}
